### ISSUE TITLE
Vacu-Vore / Meatspiking

### ISSUE THREAD

**Issue by Leon9621** [citadel-station/citadel-main#2225]
Rather simple bug. If you resist out of the aggressive grab needed to either vore someone or slam them on a meatspike, but don't interrupt the action, when it ends it will teleport you from wherever you are insode tge person / on a meatspike.

**Comment by ktccd** [comment]
Sounds like the meatspike/vore doesn't check if you're still grabbed when the timer for it ends?

**Comment by LetterJay** [comment]
@Verkister

**Comment by ktccd** [comment]
I handled the vore part of this on our side. Meatspike needs a simple fix on upstream, but I'm running out of time today and I got 2 days full of exams coming up next, so that'll have to wait in that case.

EDIT: To clarify, the meatspike code from upstream has the same kind of mistake, but it's unrelated to our code. They just forgot to check if the person was still in range when the waiting was over before spiking them. That's probably solved by updating it to use do_after, since that's the new standard I think?
